fusedrug/data/tokenizer/ops/modular_tokenizer_ops.py

```diff
@@ -0,0 +1,243 @@
+        pad_token: Union[str, None] = None,
+        pad_type_id: Union[int, None] = None,
+        validate_ends_with_eos: Optional[str] = "<EOS>",
+        verbose: Optional[bool] = False,
+        **kwargs: Any,
+    ) -> None:
+        """
+
+        Args:
+            tokenizer_path: full path to a directory that the tokenizer will be loaded from
+            max_size: sequences below this size will be padded, and above this size will be truncated
+            pad: a string of the pad token
+            pad_type_id: see tokenizers.Tokenizer.enable_padding() docstring
+            validate_ends_with_eos: during encoder request (a _call_ to the op) will make sure that it ends with the provided eos token, and raise exception otherwise.
+                having an eos (end of sentence) token in the end is useful for multiple scenarios, for example in a generative transformer (like T5 encoder-decoder)
+            verbose:
+        """
+        super().__init__(**kwargs)
+
+        if verbose:
+            print(f"DEBUG:FastModularTokenizer __init__ called for path {tokenizer_path}")
+
+        self._tokenizer_path = tokenizer_path
+        self._tokenizer = Tokenizer.from_file(self._tokenizer_path)
+        pad_id = self._tokenizer.token_to_id(pad_token)
+
+        if pad_token is None:
+            raise Exception(f"Could not find pad token = {pad_token} in {tokenizer_path}")
+
+        self._validate_ends_with_eos = validate_ends_with_eos
+
+        if self._validate_ends_with_eos is not None:
+            eos_id = self._tokenizer.token_to_id(self._validate_ends_with_eos)
+            if eos_id is None:
+                raise Exception(
+                    f"Could not find eos token = {validate_ends_with_eos} in {tokenizer_path}. You can disable the validation by setting validate_ends_with_eos=None"
+                )
+
+        self._pad_id = pad_id
+        self._verbose = verbose
+
+        if max_size is not None:
+            assert isinstance(max_size, int)
+            assert max_size > 0
+            assert isinstance(pad_id, int)
+
+            padding_kwargs = dict(length=max_size, pad_id=pad_id)
+            if pad_type_id is not None:
+                assert isinstance(pad_type_id, int)
+                padding_kwargs["pad_type_id"] = pad_type_id
+
+            self._tokenizer.enable_padding(direction="right", **padding_kwargs)
+
+            self._tokenizer.enable_truncation(
+                max_length=max_size,
+                direction="right",
+            )
+
+        self._max_size = max_size
+
+        if self._verbose:
+            self._debug_max_tokenized_len_encountered = defaultdict(int)
+
+    # note: use normalizer.Sequence to chain multiple normalizers
+    def set_normalizer(self, normalizer: Any) -> None:
+        # self._tokenizer.normalizer = normalizer
+        raise Exception("Not implemented")
+
+    # note: use pre_toknizers.Sequence to chain multiple pre_toknizers
```

Comment:
Are we planning to support it?
If not consider to delete them

Reply:
Deleted them - will appear in the next PR